lib/Parse/ParseDecl.cpp

```diff
@@ -7204,6 +7220,91 @@ void Parser::ParseCheckedPointerSpecifiers(DeclSpec &DS) {
         Diag(StartLoc, DiagID) << PrevSpec;
 }
 
+void Parser::ParseItypeforanySpecifier(DeclSpec &DS) {
```

Comment:
I think you should common this code with parsing for regular forall functions.   You can use formatting specifiers in clang error messages to customize parts of error message for specific cases.